Content.Shared/StatusEffect/StatusEffectsSystem.cs

```diff
@@ -107,6 +107,33 @@ private void OnHandleState(EntityUid uid, StatusEffectsComponent component, ref
             return false;
         }
 
+        public bool TryAddStatusEffect(EntityUid uid, string key, TimeSpan time, string component,
+            StatusEffectsComponent? status = null,
+            SharedAlertsComponent? alerts = null)
+        {
+            if (!Resolve(uid, ref status, false))
+                return false;
+
+            Resolve(uid, ref alerts, false);
+
+            if (TryAddStatusEffect(uid, key, time, status, alerts))
+            {
+                // Fuck this shit I hate it
+                var newComponent = (Component) _componentFactory.GetComponent(component);
+                newComponent.Owner = EntityManager.GetEntity(uid);
+
+                // If they already have the comp, we just won't bother updating anything.
+                if (!EntityManager.HasComponent(uid, _componentFactory.GetRegistration(component).Type))
+                {
+                    EntityManager.AddComponent(uid, newComponent);
+                    status.ActiveEffects[key].RelevantComponent = component;
+                }
```

Comment:
I have no idea how computationally expensive it is to create a new component instance, but I feel like `_componentFactory.GetComponent(component);` should probably be called after checking `HasComponent`, just in case the whole player base ever decides to go an a simultaneous drug binge and it impacts performance when unnecessary components are initialized.


```suggestion
                // If they already have the comp, we just won't bother updating anything.
                if (EntityManager.HasComponent(uid, _componentFactory.GetRegistration(component).Type))
                    return true;

                // Fuck this shit I hate it
                var newComponent = (Component) _componentFactory.GetComponent(component);
                newComponent.Owner = EntityManager.GetEntity(uid);
                EntityManager.AddComponent(uid, newComponent);
                status.ActiveEffects[key].RelevantComponent = component;
```

Reply:
oop, meant for that to be in the if block